Intermittent error: Fatal error rendering battery segment

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

I'm experiencing an intermittent issue where the battery segment fails to render. The issue resolves itself when the prompt is next displayed. I'm unaware of anything that triggers the issue; it appears to occur at random intervals.

Error output:
````
oh-my-posh fatal error rendering battery segment:runtime error: index out of range [0] with length 0

goroutine 11 [running]:
runtime/debug.Stack()
        C:/hostedtoolcache/windows/go/1.22.3/x64/src/runtime/debug/stack.go:24 +0x5e
github.com/jandedobbeleer/oh-my-posh/src/config.(*Segment).SetEnabled.func1()
        D:/a/oh-my-posh/oh-my-posh/src/config/segment.go:95 +0x48
panic({0x1867e40?, 0xc0000397a0?})
        C:/hostedtoolcache/windows/go/1.22.3/x64/src/runtime/panic.go:770 +0x132
github.com/jandedobbeleer/oh-my-posh/src/runtime/battery.systemGet(0x0)
        D:/a/oh-my-posh/oh-my-posh/src/runtime/battery/battery_windows.go:192 +0x997
github.com/jandedobbeleer/oh-my-posh/src/runtime/battery.systemGetAll()
        D:/a/oh-my-posh/oh-my-posh/src/runtime/battery/battery_windows.go:313 +0x76
github.com/jandedobbeleer/oh-my-posh/src/runtime/battery.Get()
        D:/a/oh-my-posh/oh-my-posh/src/runtime/battery/battery_windows_nix.go:60 +0x13
github.com/jandedobbeleer/oh-my-posh/src/runtime.(*Terminal).BatteryState(0xc000379cc0)
        D:/a/oh-my-posh/oh-my-posh/src/runtime/terminal_windows_nix.go:13 +0x88
github.com/jandedobbeleer/oh-my-posh/src/segments.(*Battery).Enabled(0xc0003bc5a0)
        D:/a/oh-my-posh/oh-my-posh/src/segments/battery.go:40 +0x62
github.com/jandedobbeleer/oh-my-posh/src/config.(*Segment).SetEnabled(0xc00008bdc0, {0x1a69a68, 0xc000379cc0})
        D:/a/oh-my-posh/oh-my-posh/src/config/segment.go:131 +0x36c
github.com/jandedobbeleer/oh-my-posh/src/config.(*Block).setEnabledSegments.func1(0x0?)
        D:/a/oh-my-posh/oh-my-posh/src/config/block.go:93 +0x5b
created by github.com/jandedobbeleer/oh-my-posh/src/config.(*Block).setEnabledSegments in goroutine 1
        D:/a/oh-my-posh/oh-my-posh/src/config/block.go:91 +0x98
````
<img width="658" alt="image" src="https://github.com/user-attachments/assets/2cb9389b-b2e1-43b2-a2d3-b832181a900d">



### Theme

A custom theme; the battery segment is

````json
{
    "type": "battery",
    "style": "powerline",
    "powerline_symbol": "\uE0B0",
    "foreground": "#193549",
    "background": "#ffeb3b",
    "background_templates": [
        "{{if eq \"Charging\" .State.String}}#40c4ff{{end}}",
        "{{if eq \"Discharging\" .State.String}}#ff5722{{end}}",
        "{{if eq \"Full\" .State.String}}#4caf50{{end}}"
    ],
    "properties": {
        "discharging_icon": "\uE231 ",
        "charging_icon": "\uE234 ",
        "charged_icon": "\uE22F ",
        "template": " {{ if not .Error }}{{ .Icon }}{{ .Percentage }}{{ end }}\uF295 "
    }
}

````

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
Version: 23.13.4

Shell: pwsh (7.4.5)

Prompt:

εé╢ ∩à║ εé┤ D:Γû╕sourceΓû╕Repo εé░ ε£î 8.0.400 εé░ Repo εé░ ∩Éê εéá≡ƒÜÇ 14-migrate-to-net-8 Γëí ∩üå +2 ~2 ε¡ï 1 εé░ εê» 100∩èò εé░
εé╢ ≤░¿è pwsh εé░ Name@LT-host εé░ εê║ εé░ 

Segments:

ConsoleTitle(true)                        -   4 ms
Os(true)                                  -   0 ms
Path(true)                                -   4 ms
Dotnet(true)                              - 173 ms
Project(true)                             -   2 ms
Git(true)                                 - 171 ms
Battery(true)                             - 105 ms
Shell(true)                               -   0 ms
Session(true)                             -   0 ms
Exit(true)                                -   1 ms

Run duration: 189.2623ms

Cache path: C:\Users\user\AppData\Local\oh-my-posh

Config path: C:\Users\user\.poshthemes\customtheme.omp.json

Logs:

[38;2;191;207;240m[DEBUG] [38;2;156;231;201m11:09:09.118 [0mdebug.go:49[38;2;156;231;201m ΓåÆ[0m debug mode enabled
[DEBUG] 11:09:09.119 debug.go:49 ΓåÆ plain mode enabled
[DEBUG] 11:09:09.119 terminal.go:Getenv:164 ΓåÆ C:\Users\user\AppData\Local
[TRACE] 11:09:09.119 terminal.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 11:09:09.120 terminal.go:CachePath() - 534.5┬╡s
[DEBUG] 11:09:09.120 terminal.go:70 ΓåÆ loading cache file: C:\Users\user\AppData\Local\oh-my-posh\omp.cache
[DEBUG] 11:09:09.120 terminal.go:70 ΓåÆ loading cache key: upgrade_check
[TRACE] 11:09:09.120 terminal.go(C:\Users\user\AppData\Local\oh-my-posh\omp.cache) - 0s
[DEBUG] 11:09:09.120 terminal.go:Getenv:164 ΓåÆ C:\Users\user\AppData\Local
[TRACE] 11:09:09.120 terminal.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 11:09:09.120 terminal.go:CachePath() - 532.8┬╡s
[DEBUG] 11:09:09.120 terminal.go:70 ΓåÆ loading cache file: C:\Users\user\AppData\Local\oh-my-posh\omp.cache.68940
[DEBUG] 11:09:09.120 terminal.go:70 ΓåÆ loading cache key: prompt_count_cache
[TRACE] 11:09:09.120 terminal.go(C:\Users\user\AppData\Local\oh-my-posh\omp.cache.68940) - 0s
[TRACE] 11:09:09.120 terminal.go:setPromptCount() - 0s
[DEBUG] 11:09:09.120 terminal.go:Getenv:164 ΓåÆ C:\Users\user\.poshthemes\customtheme.omp.json
[TRACE] 11:09:09.120 terminal.go:Getenv(POSH_THEME) - 0s
[DEBUG] 11:09:09.120 terminal.go:ResolveConfigPath:96 ΓåÆ config set using POSH_THEME: C:\Users\user\.poshthemes\customtheme.omp.json
[TRACE] 11:09:09.120 terminal.go:ResolveConfigPath() - 0s
[TRACE] 11:09:09.120 terminal.go:Init() - 1.7079ms
[TRACE] 11:09:09.120 terminal.go:Flags() - 0s
[TRACE] 11:09:09.122 load.go:loadConfig() - 1.4763ms
[TRACE] 11:09:09.122 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.122 debug.go:57 ΓåÆ terminal program: Windows Terminal
[DEBUG] 11:09:09.122 debug.go:57 ΓåÆ terminal shell: shell
[DEBUG] 11:09:09.122 terminal.go:Getenv:164 ΓåÆ NO DATA
[TRACE] 11:09:09.122 terminal.go:Getenv(OMP_CACHE_DISABLED) - 0s
[TRACE] 11:09:09.122 terminal_windows.go:WindowsRegistryKeyValue(HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM\ColorizationColor) - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.122 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.122 terminal_windows.go:WindowsRegistryKeyValue:215 ΓåÆ ColorizationColor(DWORD): 0xC40078D4
[TRACE] 11:09:09.122 terminal.go:Shell() - 0s
[DEBUG] 11:09:09.122 terminal.go:Getenv:164 ΓåÆ 7.4.5
[TRACE] 11:09:09.122 terminal.go:Getenv(POSH_SHELL_VERSION) - 0s
[DEBUG] 11:09:09.122 debug.go:PrintDebug:23 ΓåÆ segment: Title
[DEBUG] 11:09:09.122 text.go:Render:80 ΓåÆ rendering template: {{ .Shell }} in {{ .Folder }}
[TRACE] 11:09:09.122 terminal.go:Flags() - 0s
[TRACE] 11:09:09.123 terminal_windows.go:Root() - 239.3┬╡s
[TRACE] 11:09:09.123 terminal.go:Shell() - 0s
[TRACE] 11:09:09.123 terminal.go:StatusCodes() - 0s
[TRACE] 11:09:09.123 terminal_windows.go:IsWsl() - 0s
[DEBUG] 11:09:09.126 terminal.go:TemplateCache:651 ΓåÆ environment: [ALLUSERSPROFILE=C:\ProgramData APPDATA=C:\Users\user\AppData\Roaming chocolateyinstall=C:\Users\user\AppData\Local\UniGetUI\Chocolatey CommonProgramFiles=C:\Program Files\Common Files CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files CommonProgramW6432=C:\Program Files\Common Files COMPUTERNAME=LT-host ComSpec=C:\Windows\system32\cmd.exe CONDA_PROMPT_MODIFIER=False CustomThemeFileName=C:\Users\user\.poshthemes\customtheme.omp.json CustomThemePath=C:\Users\user\.poshthemes\ DriverData=C:\Windows\System32\Drivers\DriverData GitWorkingCopyDir=D:\source HOMEDRIVE=T: HOMEPATH=\ HOMESHARE=\\server\FIAS\StaffHomeDrives\user LOCALAPPDATA=C:\Users\user\AppData\Local LOGONSERVER=\\FIASDC NPM_CONFIG_CACHE=D:\packages\npm NUGET_PACKAGES=D:\packages\NuGet\cache NUMBER_OF_PROCESSORS=20 OneDrive=C:\Users\user\OneDrive - Company OneDriveCommercial=C:\Users\user\OneDrive - Company OneDriveConsumer=C:\Users\user\OneDrive OS=Windows_NT Path=C:\Program Files\PowerShell\7;C:\Program Files\Python312\Scripts\;C:\Program Files\Python312\;C:\Program Files (x86)\Common Files\Oracle\Java\java8path;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Program Files (x86)\VMware\VMware Workstation\bin\;C:\Program Files (x86)\Business Objects\Common\3.5\bin\NOTES\;C:\Program Files (x86)\Business Objects\Common\3.5\bin\NOTES\DATA\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\Java\jre7\bin;C:\Program Files (x86)\Java\jre7\bin;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files\Microsoft SQL Server\150\Tools\Binn\;C:\Program Files\dotnet\;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Program Files\Microsoft VS Code\bin;C:\Program Files (x86)\oh-my-posh\bin;C:\Program Files\PuTTY\;C:\Program Files (x86)\Microsoft SQL Server\160\Tools\Binn\;C:\Program Files\Microsoft SQL Server\160\Tools\Binn\;C:\Program Files\Microsoft SQL Server\160\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\160\DTS\Binn\;C:\Program Files\WinMerge;C:\Program Files (x86)\Company\Active Directory User Modification\;C:\Program Files\PowerShell\7\;C:\Program Files\WinGet\Links;C:\Program Files\GitHub CLI\;C:\Program Files\Git\cmd;C:\Users\user\AppData\Local\UniGetUI\Chocolatey\bin;C:\Users\user\AppData\Local\Programs\WingetUI\choco-cli\bin;C:\Users\user\AppData\Local\Microsoft\WindowsApps;C:\Users\user\.dotnet\tools;C:\Users\user\AppData\Local\GitHubDesktop\bin;C:\Users\user\AppData\Local\Programs\Fiddler;C:\Users\user\AppData\Local\JetBrains\Toolbox\scripts;C:\Users\user\OneDrive - Company\Apps\devtoys.cli;C:\Users\user\.dotnet\tools PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.PY;.PYW;.CPL PIP_CACHE_DIR=D:\packages\pip\cache POSH_CURSOR_COLUMN=1 POSH_CURSOR_LINE=40 POSH_GIT_ENABLED=True POSH_INSTALLER=winget POSH_PID=68940 POSH_SHELL_VERSION=7.4.5 POSH_THEME=C:\Users\user\.poshthemes\customtheme.omp.json POSH_THEMES_PATH=C:\Program Files (x86)\oh-my-posh\themes POWERLINE_COMMAND=oh-my-posh POWERSHELL_DISTRIBUTION_CHANNEL=MSI:Windows 10 Pro PROCESSOR_ARCHITECTURE=AMD64 PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 186 Stepping 2, GenuineIntel PROCESSOR_LEVEL=6 PROCESSOR_REVISION=ba02 ProgramData=C:\ProgramData ProgramFiles=C:\Program Files ProgramFiles(x86)=C:\Program Files (x86) ProgramW6432=C:\Program Files PSModulePath=D:\source\Scripts\Modules;C:\Users\user\OneDrive - Company\Documents\PowerShell\Modules;C:\Program Files\PowerShell\Modules;c:\program files\powershell\7\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\Windows\system32\WindowsPowerShell\v1.0\Modules;C:\Program Files (x86)\Microsoft SQL Server\160\Tools\PowerShell\Modules\ PUBLIC=C:\Users\Public SESSIONNAME=Console SystemDrive=C: SystemRoot=C:\Windows TEMP=C:\Users\JASON~1.MOO\AppData\Local\Temp TMP=C:\Users\JASON~1.MOO\AppData\Local\Temp UATDATA=C:\Windows\CCM\UATData\D9F8C395-CAB8-491d-B8AC-179A1FE1BE77 USERDNSDOMAIN=MY.DOMAIN USERDOMAIN=DOMAIN USERDOMAIN_ROAMINGPROFILE=DOMAIN USERNAME=user USERPROFILE=C:\Users\user windir=C:\Windows WIX=C:\Program Files (x86)\WiX Toolset v3.14\ WSLENV=WT_SESSION:WT_PROFILE_ID: WT_PROFILE_ID={574e775e-4f2a-5b96-ac1e-a2962a402336} WT_SESSION=e81e45ed-9dbb-4a39-93e8-2d7855c8fc79]
[TRACE] 11:09:09.126 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.126 terminal.go:PathSeparator() - 0s
[TRACE] 11:09:09.126 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.126 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.127 terminal.go:Pwd:189 ΓåÆ D:\source\Repo
[TRACE] 11:09:09.127 terminal.go:Pwd() - 1.029ms
[DEBUG] 11:09:09.127 terminal_windows.go:58 ΓåÆ T:\
[TRACE] 11:09:09.127 terminal_windows.go:IsWsl() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.127 terminal.go:User:330 ΓåÆ user
[TRACE] 11:09:09.127 terminal.go:User() - 0s
[DEBUG] 11:09:09.127 terminal.go:Host:347 ΓåÆ LT-host
[TRACE] 11:09:09.127 terminal.go:Host() - 0s
[TRACE] 11:09:09.127 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.127 terminal.go:Getenv:164 ΓåÆ NO DATA
[TRACE] 11:09:09.127 terminal.go:Getenv(SHLVL) - 0s
[TRACE] 11:09:09.127 terminal.go:TemplateCache() - 3.9846ms
[TRACE] 11:09:09.127 terminal.go:Shell() - 0s
[DEBUG] 11:09:09.127 terminal.go:Getenv:164 ΓåÆ 40
[TRACE] 11:09:09.127 terminal.go:Getenv(POSH_CURSOR_LINE) - 540.1┬╡s
[DEBUG] 11:09:09.127 terminal.go:Getenv:164 ΓåÆ 1
[TRACE] 11:09:09.127 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 0s
[TRACE] 11:09:09.128 terminal.go:Flags() - 0s
[TRACE] 11:09:09.128 terminal.go:Flags() - 0s
[TRACE] 11:09:09.128 terminal.go:Flags() - 0s
[TRACE] 11:09:09.128 terminal.go:Flags() - 0s
[TRACE] 11:09:09.128 terminal.go:Flags() - 0s
[TRACE] 11:09:09.128 terminal.go:Flags() - 0s
[TRACE] 11:09:09.128 terminal.go:Pwd() - 0s
[TRACE] 11:09:09.128 terminal.go:Pwd() - 0s
[TRACE] 11:09:09.128 terminal.go:Pwd() - 0s
[TRACE] 11:09:09.128 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.128 terminal_windows.go:58 ΓåÆ T:\
[TRACE] 11:09:09.128 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.128 terminal_windows.go:58 ΓåÆ T:\
[DEBUG] 11:09:09.128 segment.go:SetEnabled:115 ΓåÆ segment: Battery
[TRACE] 11:09:09.128 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.128 terminal_windows.go:58 ΓåÆ T:\
[TRACE] 11:09:09.128 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.128 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.128 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.128 segment.go:SetEnabled:115 ΓåÆ segment: Path
[DEBUG] 11:09:09.128 segment.go:SetEnabled:115 ΓåÆ segment: Os
[DEBUG] 11:09:09.128 terminal_windows.go:58 ΓåÆ T:\
[TRACE] 11:09:09.128 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.128 terminal_windows.go:58 ΓåÆ T:\
[TRACE] 11:09:09.128 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.128 segment.go:SetEnabled:115 ΓåÆ segment: Project
[TRACE] 11:09:09.128 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.128 segment.go:SetEnabled:115 ΓåÆ segment: Git
[TRACE] 11:09:09.128 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.128 path.go:149 ΓåÆ display_cygpath: false
[DEBUG] 11:09:09.128 os.go:Enabled:34 ΓåÆ ∩à║
[TRACE] 11:09:09.128 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.128 terminal.go:TemplateCache() - 0s
[TRACE] 11:09:09.128 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.128 terminal.go:PathSeparator() - 0s
[TRACE] 11:09:09.128 terminal.go:Pwd() - 0s
[TRACE] 11:09:09.128 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.128 terminal.go:Shell() - 0s
[TRACE] 11:09:09.128 terminal.go:Flags() - 0s
[TRACE] 11:09:09.128 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.128 terminal.go:Flags() - 0s
[TRACE] 11:09:09.128 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.128 terminal_windows.go:58 ΓåÆ T:\
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.129 segment.go:SetEnabled:115 ΓåÆ segment: Dotnet
[DEBUG] 11:09:09.129 path.go:setMappedLocations:563 ΓåÆ mapped_locations_enabled: true
[DEBUG] 11:09:09.129 path.go:setMappedLocations:564 ΓåÆ ∩Çô
[DEBUG] 11:09:09.129 path.go:setMappedLocations:565 ΓåÆ ∩Çô
[DEBUG] 11:09:09.129 terminal_windows.go:58 ΓåÆ T:\
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.129 terminal.go:PathSeparator() - 0s
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.129 language.go:Enabled:109 ΓåÆ extensions: [*.cs *.csx *.vb *.sln *.slnf *.csproj *.vbproj *.fs *.fsx *.fsproj global.json]
[DEBUG] 11:09:09.129 language.go:Enabled:110 ΓåÆ folders: []
[DEBUG] 11:09:09.129 path.go:setMappedLocations:571 ΓåÆ mapped_locations: map[]
[DEBUG] 11:09:09.129 language.go:Enabled:117 ΓåÆ home_enabled: false
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.129 terminal.go:Pwd() - 0s
[TRACE] 11:09:09.129 terminal.go:PathSeparator() - 0s
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.129 terminal_windows.go:58 ΓåÆ T:\
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.129 language.go:Enabled:129 ΓåÆ files
[TRACE] 11:09:09.129 terminal.go:PathSeparator() - 0s
[TRACE] 11:09:09.129 terminal.go:Pwd() - 0s
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.129 path.go:makeFolderFormatMap:839 ΓåÆ NO DATA
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.129 path.go:setStyle:218 ΓåÆ agnoster_full
[DEBUG] 11:09:09.129 path.go:colorizePath:740 ΓåÆ cycle: []
[DEBUG] 11:09:09.129 path.go:getFolderSeparator:273 ΓåÆ NO DATA
[DEBUG] 11:09:09.129 path.go:getFolderSeparator:275 ΓåÆ Γû╕
[DEBUG] 11:09:09.129 path.go:colorizePath:743 ΓåÆ cycle_folder_separator: false
[DEBUG] 11:09:09.129 path.go:colorizePath:744 ΓåÆ %s
[DEBUG] 11:09:09.129 path.go:colorizePath:746 ΓåÆ %s
[DEBUG] 11:09:09.129 path.go:colorizePath:747 ΓåÆ %s
[DEBUG] 11:09:09.129 path.go:colorizePath:748 ΓåÆ %s
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.129 terminal.go:Pwd() - 0s
[TRACE] 11:09:09.129 terminal.go:TemplateCache() - 0s
[TRACE] 11:09:09.129 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.129 terminal.go:StackCount() - 0s
[DEBUG] 11:09:09.129 terminal.go:HasFilesInDir:242 ΓåÆ false
[TRACE] 11:09:09.129 terminal.go:HasFilesInDir(package.json) - 1.1731ms
[TRACE] 11:09:09.129 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.129 terminal.go:HasFilesInDir:242 ΓåÆ false
[TRACE] 11:09:09.129 terminal.go:HasFilesInDir(Cargo.toml) - 0s
[TRACE] 11:09:09.129 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.129 terminal.go:HasFilesInDir:242 ΓåÆ false
[TRACE] 11:09:09.129 terminal.go:HasFilesInDir(pyproject.toml) - 0s
[TRACE] 11:09:09.129 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.129 terminal.go:HasFilesInDir:242 ΓåÆ false
[TRACE] 11:09:09.129 terminal.go:HasFilesInDir(composer.json) - 0s
[TRACE] 11:09:09.129 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.129 terminal.go:HasFilesInDir:242 ΓåÆ false
[TRACE] 11:09:09.129 terminal.go:HasFilesInDir(*.nuspec) - 0s
[TRACE] 11:09:09.129 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.129 terminal.go:HasFilesInDir:237 ΓåÆ true
[TRACE] 11:09:09.129 terminal.go:HasFilesInDir(*.sln) - 0s
[TRACE] 11:09:09.129 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.130 terminal.go:HasFilesInDir:242 ΓåÆ false
[TRACE] 11:09:09.130 terminal.go:HasFilesInDir(*.cs) - 1.0798ms
[TRACE] 11:09:09.130 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.130 terminal.go:HasFilesInDir:242 ΓåÆ false
[TRACE] 11:09:09.130 terminal.go:HasFilesInDir(*.csx) - 0s
[TRACE] 11:09:09.130 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.130 terminal.go:HasFilesInDir:242 ΓåÆ false
[TRACE] 11:09:09.130 terminal.go:HasFilesInDir(*.vb) - 0s
[TRACE] 11:09:09.130 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.130 terminal.go:HasFilesInDir:237 ΓåÆ true
[TRACE] 11:09:09.130 terminal.go:HasFilesInDir(*.sln) - 0s
[DEBUG] 11:09:09.130 language.go:Enabled:148 ΓåÆ fetch_version: true
[DEBUG] 11:09:09.130 language.go:setVersion:200 ΓåÆ cache_version: false
[DEBUG] 11:09:09.130 terminal.go:Getenv:164 ΓåÆ C:\Users\user\AppData\Local
[TRACE] 11:09:09.130 terminal.go:Getenv(LOCALAPPDATA) - 0s
[DEBUG] 11:09:09.130 terminal.go:LsDir:312 ΓåÆ [d .git/ - .gitignore d .vs/ - Repo.sln - Repo.sln.DotSettings.user - README.md d sample/ d src/]
[TRACE] 11:09:09.130 terminal.go:LsDir(D:\source\Repo) - 1.122ms
[TRACE] 11:09:09.130 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.130 terminal.go:FileContent:301 Γåô
    ∩╗┐
    Microsoft Visual Studio Solution File, Format Version 12.00
    # Visual Studio Version 17
    VisualStudioVersion = 17.8.34330.188
    MinimumVisualStudioVersion = 10.0.40219.1
    Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Repo", "src\Repo\Repo.csproj", "{E42DD6DA-49F0-4E67-ADAA-2D0087EFCC30}"
    EndProject
    Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ApplicationDemo", "sample\ApplicationDemo\ApplicationDemo.csproj", "{632F7241-85DE-4D81-B0C0-C3F288656397}"
    EndProject
    Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Fias.Extensions.Library", "src\Fias.Extensions.Library\Fias.Extensions.Library.csproj", "{4C8B175C-E58D-42A5-9B4B-0DF0DF4B0ABD}"
    EndProject
    Global
    	GlobalSection(SolutionConfigurationPlatforms) = preSolution
    		Debug|Any CPU = Debug|Any CPU
    		Release|Any CPU = Release|Any CPU
    	EndGlobalSection
    	GlobalSection(ProjectConfigurationPlatforms) = postSolution
    		{E42DD6DA-49F0-4E67-ADAA-2D0087EFCC30}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
    		{E42DD6DA-49F0-4E67-ADAA-2D0087EFCC30}.Debug|Any CPU.Build.0 = Debug|Any CPU
    		{E42DD6DA-49F0-4E67-ADAA-2D0087EFCC30}.Release|Any CPU.ActiveCfg = Release|Any CPU
    		{E42DD6DA-49F0-4E67-ADAA-2D0087EFCC30}.Release|Any CPU.Build.0 = Release|Any CPU
    		{632F7241-85DE-4D81-B0C0-C3F288656397}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
    		{632F7241-85DE-4D81-B0C0-C3F288656397}.Debug|Any CPU.Build.0 = Debug|Any CPU
    		{632F7241-85DE-4D81-B0C0-C3F288656397}.Release|Any CPU.ActiveCfg = Release|Any CPU
    		{632F7241-85DE-4D81-B0C0-C3F288656397}.Release|Any CPU.Build.0 = Release|Any CPU
    		{4C8B175C-E58D-42A5-9B4B-0DF0DF4B0ABD}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
    		{4C8B175C-E58D-42A5-9B4B-0DF0DF4B0ABD}.Debug|Any CPU.Build.0 = Debug|Any CPU
    		{4C8B175C-E58D-42A5-9B4B-0DF0DF4B0ABD}.Release|Any CPU.ActiveCfg = Release|Any CPU
    		{4C8B175C-E58D-42A5-9B4B-0DF0DF4B0ABD}.Release|Any CPU.Build.0 = Release|Any CPU
    	EndGlobalSection
    	GlobalSection(SolutionProperties) = preSolution
    		HideSolutionNode = FALSE
    	EndGlobalSection
    	GlobalSection(ExtensibilityGlobals) = postSolution
    		SolutionGuid = {FD887F86-A52E-4ABB-827F-B56550DC1A7E}
    	EndGlobalSection
    EndGlobal
    
[TRACE] 11:09:09.130 terminal.go:FileContent(Repo.sln) - 0s
[ERROR] 11:09:09.131 project.go:getDotnetProject:244 ΓåÆ cannot extract TFM from Repo project file
[TRACE] 11:09:09.131 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.132 win32_windows.go:isWriteable:303 ΓåÆ current user is member of S-1-5-32-544
[DEBUG] 11:09:09.132 win32_windows.go:isWriteable:311 Γåô
    WRITE_DAC
    WRITE_OWNER
    SYNCHRONIZE
    DELETE
    READ_CONTROL
[DEBUG] 11:09:09.132 win32_windows.go:isWriteable:313 ΓåÆ user has write access
[TRACE] 11:09:09.132 terminal_windows.go:DirIsWritable() - 3.8093ms
[TRACE] 11:09:09.132 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.149 terminal.go:CommandPath:389 ΓåÆ C:\Program Files\dotnet\dotnet.exe
[TRACE] 11:09:09.149 terminal.go:CommandPath(dotnet) - 19.4698ms
[TRACE] 11:09:09.149 terminal.go:HasCommand(dotnet) - 19.4698ms
[DEBUG] 11:09:09.158 terminal.go:CommandPath:389 ΓåÆ C:\Program Files\Git\cmd\git.exe
[TRACE] 11:09:09.158 terminal.go:CommandPath(git.exe) - 29.9079ms
[TRACE] 11:09:09.158 terminal.go:HasCommand(git.exe) - 29.9079ms
[DEBUG] 11:09:09.158 git.go:shouldDisplay:312 ΓåÆ fetch_bare_info: false
[TRACE] 11:09:09.158 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.158 terminal.go:ResolveSymlink:286 ΓåÆ D:\source\Repo
[TRACE] 11:09:09.158 terminal.go:ResolveSymlink(D:\source\Repo) - 151.7┬╡s
[TRACE] 11:09:09.158 terminal.go:HasParentFilePath(.git) - 151.7┬╡s
[DEBUG] 11:09:09.158 scm.go:shouldIgnoreRootRepository:150 ΓåÆ exclude_folders: []
[DEBUG] 11:09:09.158 terminal_windows.go:58 ΓåÆ T:\
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.158 git.go:Enabled:176 ΓåÆ fetch_user: false
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.158 git.go:Enabled:191 ΓåÆ pwsh
[DEBUG] 11:09:09.158 terminal.go:Getenv:164 ΓåÆ NO DATA
[TRACE] 11:09:09.158 terminal.go:Getenv(POSH_GIT_STATUS) - 0s
[ERROR] 11:09:09.158 posh_git.go:hasPoshGitStatus:49 ΓåÆ POSH_GIT_STATUS environment variable not set, do you have the posh-git module installed?
[DEBUG] 11:09:09.158 git.go:Enabled:196 ΓåÆ fetch_status: true
[DEBUG] 11:09:09.158 git.go:shouldIgnoreStatus:454 ΓåÆ ignore_status: []
[DEBUG] 11:09:09.158 terminal_windows.go:58 ΓåÆ T:\
[TRACE] 11:09:09.158 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.158 git.go:setGitStatus:600 ΓåÆ status_formats: map[]
[DEBUG] 11:09:09.158 git.go:getSwitchMode:899 ΓåÆ untracked_modes: map[]
[DEBUG] 11:09:09.158 git.go:getSwitchMode:899 ΓåÆ ignore_submodules: map[]
[TRACE] 11:09:09.234 terminal_windows_nix.go:BatteryState() - 105.9884ms
[DEBUG] 11:09:09.234 battery.go:Enabled:59 ΓåÆ εê» 
[TRACE] 11:09:09.234 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.298 terminal.go:RunCommand:367 Γåô
    # branch.oid ec5bb6b03ac22d78019761edc388c62c13b51706
    # branch.head feat/14-migrate-to-net-8
    # branch.upstream origin/feat/14-migrate-to-net-8
    # branch.ab +0 -0
    1 M. N... 100644 100644 100644 511683f84a85c47038789e9e1633375ecfd793c9 4fa36a5e2e767c6596078374d9156384be03a5c9 sample/ApplicationDemo/ApplicationDemo.csproj
    1 A. N... 000000 100644 100644 0000000000000000000000000000000000000000 ff907e4d8162932cea5b90260639b7da50c787b6 sample/ApplicationDemo/Borrower.cs
    1 A. N... 000000 100644 100644 0000000000000000000000000000000000000000 a81b91b200354725f3fab6c32ba205c852b0915c sample/ApplicationDemo/Display.cs
    1 M. N... 100644 100644 100644 4f99c7b0660cd4bda09fc102a9607cffb0984f2e e1fe3584c015a2d1fdb4d655c42b55d27348ec27 sample/ApplicationDemo/Program.cs
[TRACE] 11:09:09.298 terminal.go:RunCommand(git.exe -C D:/source/Repo --no-optional-locks -c core.quotepath=false -c color.status=false status -unormal --branch --porcelain=2) - 139.4318ms
[DEBUG] 11:09:09.298 git.go:setGitHEADContext:659 ΓåÆ εéá
[DEBUG] 11:09:09.299 scm.go:formatBranch:114 ΓåÆ mapped_branches: map[bug/*:≡ƒÉ¢  feat/*:≡ƒÜÇ  issue/*:≡ƒÆÑ ]
[DEBUG] 11:09:09.299 scm.go:formatBranch:130 ΓåÆ full_branch_path: true
[DEBUG] 11:09:09.299 scm.go:formatBranch:136 ΓåÆ branch_max_length: 0
[DEBUG] 11:09:09.299 terminal.go:HasFolder:271 ΓåÆ false
[TRACE] 11:09:09.299 terminal.go:HasFolder(D:\source\Repo\.git/rebase-merge) - 0s
[DEBUG] 11:09:09.299 terminal.go:HasFolder:271 ΓåÆ false
[TRACE] 11:09:09.299 terminal.go:HasFolder(D:\source\Repo\.git/rebase-apply) - 0s
[DEBUG] 11:09:09.299 git.go:setGitHEADContext:710 ΓåÆ ∩Éù
[DEBUG] 11:09:09.299 terminal.go:HasFilesInDir:242 ΓåÆ false
[TRACE] 11:09:09.299 terminal.go:HasFilesInDir(MERGE_MSG) - 0s
[DEBUG] 11:09:09.299 terminal.go:HasFilesInDir:242 ΓåÆ false
[TRACE] 11:09:09.299 terminal.go:HasFilesInDir(CHERRY_PICK_HEAD) - 0s
[DEBUG] 11:09:09.299 terminal.go:HasFilesInDir:242 ΓåÆ false
[TRACE] 11:09:09.299 terminal.go:HasFilesInDir(REVERT_HEAD) - 0s
[DEBUG] 11:09:09.299 terminal.go:HasFilesInDir:242 ΓåÆ false
[TRACE] 11:09:09.299 terminal.go:HasFilesInDir(sequencer/todo) - 0s
[DEBUG] 11:09:09.299 git.go:473 ΓåÆ Γëí
[DEBUG] 11:09:09.299 git.go:Enabled:207 ΓåÆ fetch_upstream_icon: true
[DEBUG] 11:09:09.299 git.go:getUpstreamIcon:537 ΓåÆ upstream_icons: map[custom.domain:Γÿó∩╕Å ]
[DEBUG] 11:09:09.299 git.go:getUpstreamIcon:558 ΓåÆ ∩Éê 
[TRACE] 11:09:09.299 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.302 terminal.go:RunCommand:367 ΓåÆ 8.0.400
[TRACE] 11:09:09.302 terminal.go:RunCommand(dotnet --version) - 152.5851ms
[DEBUG] 11:09:09.302 language.go:buildVersionURL:280 ΓåÆ https://github.com/dotnet/core/blob/master/release-notes/{{ .Major }}.{{ .Minor }}/{{ .Major }}.{{ .Minor }}.{{ substr 0 1 .Patch }}/{{ .Major }}.{{ .Minor }}.{{ substr 0 1 .Patch }}.md
[DEBUG] 11:09:09.302 text.go:Render:80 ΓåÆ rendering template: https://github.com/dotnet/core/blob/master/release-notes/{{ .Major }}.{{ .Minor }}/{{ .Major }}.{{ .Minor }}.{{ substr 0 1 .Patch }}/{{ .Major }}.{{ .Minor }}.{{ substr 0 1 .Patch }}.md
[TRACE] 11:09:09.302 terminal.go:Flags() - 0s
[TRACE] 11:09:09.302 terminal.go:TemplateCache() - 0s
[TRACE] 11:09:09.302 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.302 text.go:Render:80 ΓåÆ rendering template:  {{ if not .Error }}{{ .Icon }}{{ .Percentage }}{{ end }}∩èò 
[TRACE] 11:09:09.302 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.302 text.go:Render:80 ΓåÆ rendering template:  ε£î {{ if .Unsupported }}∩ü▒{{ else }}{{ .Full }}{{ end }} 
[TRACE] 11:09:09.302 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.302 text.go:Render:80 ΓåÆ rendering template:  {{ if .WSL }}WSL at {{ end }}{{.Icon}} 
[TRACE] 11:09:09.302 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.302 text.go:Render:80 ΓåÆ rendering template:  {{ .Path }} 
[TRACE] 11:09:09.302 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.302 text.go:Render:80 ΓåÆ rendering template:  {{ .UpstreamIcon }}{{ .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed }} ∩üä {{ .Working.String }}{{ end }}{{ if and (.Working.Changed) (.Staging.Changed) }} |{{ end }}{{ if .Staging.Changed }} ∩üå {{ .Staging.String }}{{ end }}{{ if gt .StashCount 0 }} ε¡ï {{ .StashCount }}{{ end }} 
[TRACE] 11:09:09.302 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.302 text.go:Render:80 ΓåÆ rendering template: {{ if .Error }}{{ .Error }}{{ else }}{{ if .Version }}∩Æç {{.Version}}{{ end }} {{ if .Name }}{{ .Name }}{{ end }}{{ end }} 
[TRACE] 11:09:09.302 terminal.go:Flags() - 0s
[TRACE] 11:09:09.302 terminal.go:TemplateCache() - 0s
[TRACE] 11:09:09.302 terminal.go:TemplateCache() - 0s
[TRACE] 11:09:09.303 terminal.go:TemplateCache() - 0s
[TRACE] 11:09:09.303 terminal.go:TemplateCache() - 0s
[TRACE] 11:09:09.303 terminal.go:TemplateCache() - 0s
[TRACE] 11:09:09.303 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.303 terminal.go:FileContent:301 Γåô
    0000000000000000000000000000000000000000 a042a521a5cebcf5c27108e400fe0cb16f38b869 Jason Moore <145747752+FIASJason@users.noreply.github.com> 1726627413 +0800	WIP on feat/14-migrate-to-net-8: b6bb44a Update version.
    
[TRACE] 11:09:09.303 terminal.go:FileContent(D:\source\Repo\.git/logs/refs/stash) - 0s
[DEBUG] 11:09:09.304 text.go:Render:80 ΓåÆ rendering template: diamond
[TRACE] 11:09:09.304 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.304 text.go:Render:80 ΓåÆ rendering template: powerline
[TRACE] 11:09:09.304 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.304 text.go:Render:80 ΓåÆ rendering template: powerline
[TRACE] 11:09:09.304 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.304 text.go:Render:80 ΓåÆ rendering template: powerline
[TRACE] 11:09:09.304 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.304 text.go:Render:80 ΓåÆ rendering template: {{ if or (.Working.Changed) (.Staging.Changed) }}#aaffc3{{ end }}
[TRACE] 11:09:09.304 terminal.go:Flags() - 0s
[TRACE] 11:09:09.304 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.304 text.go:Render:80 ΓåÆ rendering template: powerline
[TRACE] 11:09:09.304 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.304 text.go:Render:80 ΓåÆ rendering template: {{ if or (.Working.Changed) (.Staging.Changed) }}#aaffc3{{ end }}
[TRACE] 11:09:09.304 terminal.go:Flags() - 0s
[TRACE] 11:09:09.304 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.304 text.go:Render:80 ΓåÆ rendering template: {{if eq "Charging" .State.String}}#40c4ff{{end}}
[TRACE] 11:09:09.304 terminal.go:Flags() - 0s
[TRACE] 11:09:09.304 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.304 text.go:Render:80 ΓåÆ rendering template: {{if eq "Discharging" .State.String}}#ff5722{{end}}
[TRACE] 11:09:09.304 terminal.go:Flags() - 0s
[TRACE] 11:09:09.304 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.304 text.go:Render:80 ΓåÆ rendering template: {{if eq "Full" .State.String}}#4caf50{{end}}
[TRACE] 11:09:09.304 terminal.go:Flags() - 0s
[TRACE] 11:09:09.304 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.304 text.go:Render:80 ΓåÆ rendering template: powerline
[TRACE] 11:09:09.304 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.304 text.go:Render:80 ΓåÆ rendering template: {{ if or (.Working.Changed) (.Staging.Changed) }}#aaffc3{{ end }}
[TRACE] 11:09:09.304 terminal.go:Flags() - 0s
[TRACE] 11:09:09.305 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.305 text.go:Render:80 ΓåÆ rendering template: {{if eq "Charging" .State.String}}#40c4ff{{end}}
[TRACE] 11:09:09.305 terminal.go:Flags() - 0s
[TRACE] 11:09:09.305 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.305 text.go:Render:80 ΓåÆ rendering template: {{if eq "Discharging" .State.String}}#ff5722{{end}}
[TRACE] 11:09:09.305 terminal.go:Flags() - 0s
[TRACE] 11:09:09.305 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.305 text.go:Render:80 ΓåÆ rendering template: {{if eq "Full" .State.String}}#4caf50{{end}}
[TRACE] 11:09:09.305 terminal.go:Flags() - 0s
[TRACE] 11:09:09.305 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.305 text.go:Render:80 ΓåÆ rendering template: {{if eq "Charging" .State.String}}#40c4ff{{end}}
[TRACE] 11:09:09.305 terminal.go:Flags() - 0s
[TRACE] 11:09:09.305 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.305 text.go:Render:80 ΓåÆ rendering template: {{if eq "Discharging" .State.String}}#ff5722{{end}}
[TRACE] 11:09:09.305 terminal.go:Flags() - 0s
[TRACE] 11:09:09.305 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.305 text.go:Render:80 ΓåÆ rendering template: {{if eq "Full" .State.String}}#4caf50{{end}}
[TRACE] 11:09:09.305 terminal.go:Flags() - 0s
[TRACE] 11:09:09.306 terminal.go:TemplateCache() - 0s
[TRACE] 11:09:09.306 terminal.go:Shell() - 0s
[TRACE] 11:09:09.306 terminal.go:Flags() - 0s
[TRACE] 11:09:09.306 terminal.go:Flags() - 0s
[TRACE] 11:09:09.306 terminal.go:Flags() - 0s
[TRACE] 11:09:09.306 terminal.go:Flags() - 0s
[TRACE] 11:09:09.306 terminal.go:Pwd() - 0s
[TRACE] 11:09:09.307 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.307 status.go:Init:53 ΓåÆ {{ .Code }}
[DEBUG] 11:09:09.307 terminal_windows.go:58 ΓåÆ T:\
[TRACE] 11:09:09.307 terminal.go:Pwd() - 0s
[DEBUG] 11:09:09.307 terminal_windows.go:58 ΓåÆ T:\
[TRACE] 11:09:09.307 terminal.go:GOOS() - 0s
[TRACE] 11:09:09.307 terminal.go:Pwd() - 0s
[TRACE] 11:09:09.307 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.307 segment.go:SetEnabled:115 ΓåÆ segment: Shell
[DEBUG] 11:09:09.307 shell.go:Enabled:28 ΓåÆ mapped_shell_names: map[]
[TRACE] 11:09:09.307 terminal.go:Shell() - 0s
[TRACE] 11:09:09.307 terminal.go:Flags() - 0s
[TRACE] 11:09:09.307 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.307 segment.go:SetEnabled:115 ΓåÆ segment: Root
[DEBUG] 11:09:09.307 terminal_windows.go:58 ΓåÆ T:\
[TRACE] 11:09:09.307 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.307 segment.go:SetEnabled:115 ΓåÆ segment: Exit
[TRACE] 11:09:09.307 terminal.go:StatusCodes() - 0s
[DEBUG] 11:09:09.307 text.go:Render:80 ΓåÆ rendering template: {{ .Code }}
[TRACE] 11:09:09.307 terminal_windows.go:Root() - 0s
[TRACE] 11:09:09.307 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.307 terminal_windows.go:58 ΓåÆ T:\
[TRACE] 11:09:09.307 terminal.go:GOOS() - 0s
[DEBUG] 11:09:09.307 segment.go:SetEnabled:115 ΓåÆ segment: Session
[DEBUG] 11:09:09.307 terminal.go:Getenv:164 ΓåÆ NO DATA
[TRACE] 11:09:09.307 terminal.go:Getenv(SSH_CONNECTION) - 0s
[DEBUG] 11:09:09.307 terminal.go:Getenv:164 ΓåÆ NO DATA
[TRACE] 11:09:09.307 terminal.go:Getenv(SSH_CLIENT) - 0s
[TRACE] 11:09:09.307 terminal.go:TemplateCache() - 0s
[TRACE] 11:09:09.307 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.307 status.go:Enabled:42 ΓåÆ always_enabled: true
[TRACE] 11:09:09.307 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.307 text.go:Render:80 ΓåÆ rendering template: <transparent></> {{ if gt .Code 0 }}∩Çì{{ else }}εê║{{ end }} 
[TRACE] 11:09:09.307 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.307 text.go:Render:80 ΓåÆ rendering template:  {{ if .SSHSession }}∩áù {{ end }}{{ .UserName }}@{{ .HostName }} 
[TRACE] 11:09:09.307 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.307 text.go:Render:80 ΓåÆ rendering template: {{ if eq .Name "pwsh" }} ≤░¿è{{else}} ∩▓╡{{end}} {{ .Name }} 
[TRACE] 11:09:09.307 terminal.go:Flags() - 0s
[TRACE] 11:09:09.307 terminal.go:TemplateCache() - 0s
[TRACE] 11:09:09.307 terminal.go:TemplateCache() - 0s
[TRACE] 11:09:09.307 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.307 text.go:Render:80 ΓåÆ rendering template: powerline
[TRACE] 11:09:09.307 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.307 text.go:Render:80 ΓåÆ rendering template: diamond
[TRACE] 11:09:09.307 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.307 text.go:Render:80 ΓåÆ rendering template: powerline
[TRACE] 11:09:09.307 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.307 text.go:Render:80 ΓåÆ rendering template: {{ if gt .Code 0 }}#f1184c{{ end }}
[TRACE] 11:09:09.307 terminal.go:Flags() - 0s
[TRACE] 11:09:09.307 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.307 text.go:Render:80 ΓåÆ rendering template: powerline
[TRACE] 11:09:09.307 terminal.go:Flags() - 0s
[DEBUG] 11:09:09.307 text.go:Render:80 ΓåÆ rendering template: {{ if gt .Code 0 }}#f1184c{{ end }}
[TRACE] 11:09:09.307 terminal.go:Flags() - 0s
[TRACE] 11:09:09.308 terminal.go:TemplateCache() - 0s
[DEBUG] 11:09:09.308 text.go:Render:80 ΓåÆ rendering template: {{ if gt .Code 0 }}#f1184c{{ end }}
[TRACE] 11:09:09.308 terminal.go:Flags() - 0s
[TRACE] 11:09:09.308 terminal.go:TemplateCache() - 0s
[TRACE] 11:09:09.308 terminal.go:Shell() - 0s
[TRACE] 11:09:09.308 terminal.go:Flags() - 0s
[TRACE] 11:09:09.308 terminal.go:Shell() - 0s
[DEBUG] 11:09:09.308 terminal.go:Getenv:164 ΓåÆ C:\Users\user\AppData\Local
[TRACE] 11:09:09.308 terminal.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 11:09:09.308 terminal.go:CachePath() - 0s
[TRACE] 11:09:09.308 terminal.go:Flags() - 0s
```

--- a/src/runtime/battery/battery_windows.go
+++ b/src/runtime/battery/battery_windows.go
@@ -185,6 +185,10 @@ func systemGet(idx int) (*battery, error) {
 		return nil, errno
 	}
 
+	if cbRequired == 0 {
+		return nil, errors.New("no buffer information returned")
+	}
+
 	// The god damn struct with ANYSIZE_ARRAY of utf16 in it is crazy.
 	// So... let's emulate it with array of uint16 ;-D.
 	// Keep in mind that the first two elements are actually cbSize.
@@ -250,7 +254,7 @@ func systemGet(idx int) (*battery, error) {
 	}
 
 	if bqi.BatteryTag == 0 {
-		return nil, errors.New("BatteryTag not returned")
+		return nil, errors.New("battery tag not returned")
 	}
 
 	b := &battery{}
@@ -274,6 +278,7 @@ func systemGet(idx int) (*battery, error) {
 	b.Full = float64(bi.FullChargedCapacity)
 
 	bws := batteryWaitStatus{BatteryTag: bqi.BatteryTag}
+
 	var bs batteryStatus
 	err = windows.DeviceIoControl(
 		handle,

